### PR TITLE
feat(app-startup): Add Cold/Warm start widgets on summary page

### PR DESCRIPTION
**Pull request by narsaynorath**
Adds a query to get cold/warm start data from metrics and then plots it into two widgets that compares two releases. I reused code from the screens charts to transform the data into different y-axes by release and device class

Screenshot 2024-01-02 at 1 11 25 PM

### CODE REVIEW

File: static/app/views/starfish/views/appStartup/screenSummary/deviceClassBreakdownBarChart.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+const XAXIS_CATEGORIES = ['high', 'medium', 'low', 'Unknown'];
+const CHART_HEIGHT = 70;
+
+interface DeviceClassBreakdownBarChartProps {
+  isError: boolean;
+  isLoading: boolean;
+  title: string;
+  yAxis: string;
+  data?: any;
```
Comment (narsaynorath) [inline static/app/views/starfish/views/appStartup/screenSummary/deviceClassBreakdownBarChart.tsx:27]: Oops. I missed this. Will add a type

File: static/app/views/starfish/views/appStartup/screenSummary/widgets.tsx [hunk 2]
```diff
@@ -1,11 +1,128 @@
...
+      {
+        name: '',
+        fields: [
+          'avg(measurements.app_start_cold)',
+          'avg(measurements.app_start_warm)',
+          'device.class',
+          'release',
+        ],
+        yAxis: YAXES.map(val => YAXIS_COLUMNS[val]),
+        topEvents: '2',
```
Comment (shruthilayaj) [inline static/app/views/starfish/views/appStartup/screenSummary/widgets.tsx:63]: nit: we can get rid of this?

File: static/app/views/starfish/views/appStartup/screenSummary/widgets.tsx [hunk 3]
```diff
@@ -1,11 +1,128 @@
...
+        data: Array(['high', 'medium', 'low', 'Unknown'].length).fill(0),
+      };
+    }
+    if (secondaryRelease) {
+      transformedData[YAXIS_COLUMNS[val]][secondaryRelease] = {
+        seriesName: secondaryRelease,
+        data: Array(['high', 'medium', 'low', 'Unknown'].length).fill(0),
+      };
+    }
+  });
```
Comment (shruthilayaj) [inline static/app/views/starfish/views/appStartup/screenSummary/widgets.tsx:99]: nit: we do this in a few places now, should we move it to a util?

**Review by shruthilayaj — APPROVED** [review]

File: static/app/views/starfish/views/appStartup/screenSummary/widgets.tsx (hunk 3 above)
Comment (narsaynorath) [reply static/app/views/starfish/views/appStartup/screenSummary/widgets.tsx]: Good point, I turned this into a separate util than the helper I made for the last time I copied code like this over.

I noticed it's generally the same so there's an opportunity to refactor and make them both work for the same purpose, but for now I'm going to keep them separate. There are a few things to generalize and I don't want to do too much in this PR that may affect the screens module.

**Pull request merged**